immich-app/immich
# feat(web,server): user memory settings

Fixes #3627

In this PR:
- Add a new user property `memoriesEnabled` to the user entity
- Disable `MemoryLane` in the web when this is set to false
- Add new `Memories` settings under User Settings

Tested Scenarios:
- Set to false, verify no memory lane
- Set to true, verify memory lane
- e2e test to update user and set memories to false
- e2e test to create user with memories set to false
- unit tests all pass

TODO:
- [x] Disable memory lane in the mobile app when turned off

![image](https://github.com/immich-app/immich/assets/4334196/95398637-9bd5-41a5-93e7-bb445d013b4e)
![image](https://github.com/immich-app/immich/assets/4334196/3948f284-9d9d-4e94-b1e2-a591db24c30f)
